.github/workflows/rust-pipeline.yml

```diff
@@ -0,0 +1,101 @@
+name: Rust pipeline
+
+on:
+  workflow_call:
+    inputs:
+      runs-on:
+        required: false
+        type: string
+        default: ubuntu-latest-16core
+
+# Environment variables for all jobs.
+env:
+  CARGO_TERM_COLOR: always
+  ADDITIONAL_PACKAGES: ""
+
+# Defined CI jobs.
+jobs:
+  check:
+    runs-on: ${{ inputs.runs-on }}
+    container: ghcr.io/famedly/rust-container:nightly
+    steps:
+    - name: Checkout current repository
+      uses: actions/checkout@v3
+      with:
+        path: head
+
+    - uses: famedly/backend-build-workflows/.github/actions/rust-prepare@main
+      with:
+        gitlab_ssh: ${{ secrets.CI_SSH_PRIVATE_KEY}}
+        gitlab_user: ${{ secrets.GITLAB_USER }}
+        gitlab_pass: ${{ secrets.GITLAB_PASS }}
+    
+    - name: Caching
+      uses: Swatinem/rust-cache@b8a6852b4f997182bdea832df3f9e153038b5191
+      with:
+        workspaces: "head -> target"
+
+    - name: Rustfmt
+      working-directory: head
+      run: cargo +${NIGHTLY_VERSION} fmt -- --check
+
+    - name: Clippy
+      working-directory: head
+      run: cargo lints clippy --workspace --all-targets -- -D warnings
+    
+    - name: Test & Coverage
+      working-directory: head
+      run: cargo llvm-cov nextest --workspace --lcov --output-path $GITHUB_WORKSPACE/new-cov.lcov
+    
+    - name: Doc-test
+      working-directory: head
+      run: cargo +${NIGHTLY_VERSION} test --doc --workspace --verbose
+
+    - name: Udeps
+      working-directory: head
+      run: cargo +${NIGHTLY_VERSION} udeps
+    
+    # - name: Typos
+    #   uses: crate-ci/typos@ef5fcf92dfbd679f97c0371159e143852f7b1eb1
+    #   with: 
+    #     files: ./head
```

Comment:
Does the action not work?

Reply:
Ahhhh, I didn't know I can approve actions by going to organization settings